### PR TITLE
Option to list rendering backends

### PR DESCRIPTION
**Pull request by Meakk**
Fix https://github.com/f3d-app/f3d/issues/1697

Add information about available rendering backends

### CODE REVIEW

**Comment by snoyer** [comment]
would need to use `--rendering-backends-list` (plural) to match `--readers-list` and `--bindings-list`
However I believe the singular is grammatically correct so the other 2 would need to change instead/eventually?

File: application/F3DOptionsTools.cxx [hunk 1]
```diff
@@ -176,7 +176,7 @@ static inline const std::array<CLIGroup, 8> CLIOptions = {{
  * True boolean options need to be filtered out in ParseCLIOptions
  * This is the easiest, compile time way to do it
  */
-constexpr std::array<std::string_view, 4> CLIBooleans = {"version", "help", "readers-list", "scan-plugins"};
+constexpr std::array CLIBooleans = {"version", "help", "readers-list", "scan-plugins", "renderin-backend-list"};
```
[the remark this answers is not in the document]
Comment (Meakk) [reply application/F3DOptionsTools.cxx]: yes since C++14

File: doc/user/CONFIGURATION_FILE.md [hunk 2]
```diff
@@ -65,7 +65,7 @@ The third block specifies raytracing usage for .gltf and .glb files.
 The last block specifies that volume rendering should be used with .mhd files.
 
 The following options <b> cannot </b> be set via config file:
-`help`, `version`, `readers-list`, `config`, `dry-run` and `input`.
+`help`, `version`, `list-readers`, `config`, `dry-run` and `input`.
```
Comment (mwestphal) [inline doc/user/CONFIGURATION_FILE.md:68]: I think `list-rendering-backends` and `scan-plugins` should be added here

File: doc/user/OPTIONS.md [hunk 3]
```diff
@@ -13,7 +13,7 @@ Options|Default|Description
 \-\-no-background||Use with \-\-output to output a png file with a transparent background.
 -h, \-\-help||Print *help* and exit. Ignore `--verbose`.
 \-\-version||Show *version* information and exit. Ignore `--verbose`.
-\-\-readers-list||List available *readers* and exit. Ignore `--verbose`.
+\-\-list-readers||List available *readers* and exit. Ignore `--verbose`.
```
Comment (mwestphal) [inline doc/user/OPTIONS.md:16]: I'd expect list-bindings and list-rendering-backends to appear in this file

**Review by mwestphal — CHANGES_REQUESTED** [review]
small changes needed

File: doc/user/OPTIONS.md [hunk 4]
```diff
@@ -13,7 +13,9 @@ Options|Default|Description
...
+\-\-list-bindings||List available *bindings* and exit. Ignore `--verbose`.
```
Comment (mwestphal) [inline doc/user/OPTIONS.md:17]: not sure that it ignore `--verbose` though

**Review by mwestphal — CHANGES_REQUESTED** [review]
small question

File: doc/user/OPTIONS.md (hunk 4 above)
Comment (Meakk) [reply doc/user/OPTIONS.md:17]: Indeed. It's copied from other list commands.

Comment (Meakk) [reply doc/user/OPTIONS.md:17]: Just checked, it's correct. The verbosity level is not taken into account yet

**Review by mwestphal — APPROVED** [review]

**Pull request merged**